server/schedule/filter/filters.go

```diff
@@ -543,12 +543,51 @@ func (f *ruleFitFilter) Target(opt opt.Options, store *core.StoreInfo) bool {
 	return placement.CompareRegionFit(f.oldFit, newFit) <= 0
 }
 
+type engineFilter struct {
+	scope      string
+	constraint placement.LabelConstraint
+}
+
+// NewEngineFilter creates a filter that filters out default engine stores.
+// By default, all stores that are not marked with a special engine will be filtered out.
+// Specify the special engine label if you want to include the special stores.
+func NewEngineFilter(scope string, allowSpeicalEngines ...string) Filter {
```

Comment:
There are too many inverse logic and it's hard to understand.

Reply:
any suggestion?